### DISCUSSION TITLE
Publish date in far future not displayed correctly

### DISCUSSION THREAD

**Discussion by http403** [FreshRSS/FreshRSS#5569] [Q&A] [answered]
Hello all,

I just migrated to FreshRSS and I discovered that the date of one of my feed isn't displayed correctly. The date is in the far future (3309) and the reason is to fit the overall theme of the feed. I would like to know if FreshRSS will ever support this before opening an issue and get a `wont fix` after.

Thanks

Feed URL: https://www.elitedangerous.com/en-GB/rss/galnet.xml
What FreshFeed display: ![image](https://github.com/FreshRSS/FreshRSS/assets/8369976/8c5ab366-95ce-40ea-b7f4-66092df0abec)

**Comment by Frenzie** [comment]
> I would like to know if FreshRSS will ever support this before opening an issue and get a wont fix after.

It seems to me this is completely identical to that? Anyway, it's clearly a [year 2038 problem](https://en.wikipedia.org/wiki/Year_2038_problem).

The code seems to be aware of the issue.
https://github.com/FreshRSS/FreshRSS/blob/8f188b57ee20f188a9f882fdf6e5cbf5efa0028d/app/SQL/install.sql.mysql.php#L27

I'm not sure if there is a specific recommended upgrade path (other than using `DATETIME` in MySQL/MariaDB or just using larger integers). Perhaps it's as simple as switching to larger integers, but perhaps the Google Reader API fundamentally suffers from the problem as well. Perhaps @Alkarex has already thought about it in the past.

**Reply by Alkarex** [reply]
Assuming PHP running on a 64-bit platform, from the top of my head, SQLite and the Google Reader API should already be fine for 2038+ (although clients will have to support it), while MariaDB/MySQL and PostgreSQL should have some date fields upgraded to `BIGINT`. I was not planning this upgrade that early (I thought something like 10 years in advance should be fine), but there might indeed already be some use-cases appearing, so we could consider it. Note that MariaDB/MySQL native `TIMESTAMP` type seems to also suffer from the same limitation at year 2038.

**Reply by http403** [reply]
Thanks for considering it. Should I create an issue form this discussion?

**Reply by Alkarex** [reply]
Would you be interested / able to test https://github.com/FreshRSS/FreshRSS/pull/5570 ?
At the moment, the update is not applied automatically on existing accounts (can be fixed), but can be tested by creating a new user after applying the patch.

**Reply by Frenzie** [reply]
> Note that MariaDB/MySQL native TIMESTAMP type seems to also suffer from the same limitation at year 2038.

That's part of why I said DATETIME, although fwiw I believe that has a y10k problem. ^_^

**Reply by http403** [reply]
@Alkarex I'm not experienced in testing so there might be errors in my testing.
Unfortunately the patch doesn't work. I tried in 3 ways, all on a LEMP stack with PHP version 8.2.8 64-bit. All sources downloaded from GH.

1. checkout 1.21.0, install, apply patch, and create new account
2. checkout 1.21.0, apply the patch, and install
3. checkout `Alkarex/FreshRSS@date-2038` commit 2d73696 , and install

All of them doesn't work and the date is still the same as the first image shown.

I checked the MySQL table schema and all of them reflected the schema update, except the first user in the first test.

**Reply by Alkarex** [reply]
Could you please try again @http403 ?

**Reply by http403** [reply]
@Alkarex I can confirm the patch works for all three methods. Patching the stable version require a bit manual editing of [EntryDAO.php](https://github.com/FreshRSS/FreshRSS/blob/0fe0ce894cbad09757d719dd4b400b9862c1a12a/app/Models/EntryDAO.php#L255) though.

**Reply by http403** [reply]
I also confirm the changes compatiable with the Google Reader API, at least for clients that can display it.

**Reply by Alkarex** [reply, marked as answer]
In conclusion, dates after year 2038 will be supported by FreshRSS 1.23.0+ (i.e. after https://github.com/FreshRSS/FreshRSS/pull/5570 has been merged)

**Reply by Alkarex** [reply]
@http403 Merged in `edge`. Tests and feedback welcome!
